domain/transfer/receiver/repository.go

```diff
@@ -154,3 +155,31 @@ func (f repository) CreateOrUpdate(ctx context.Context, status, taskID string, c
 
 	return &incomingTransfer, nil
 }
+
+func (f repository) FindByFHIRTaskID(ctx context.Context, customerID int, fhirTaskID string) (*domain.IncomingTransfer, error) {
+	tx, err := sqlUtil.GetTransaction(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	const query = `SELECT * FROM incoming_transfers WHERE customer_id = :customer_id AND task_id = :task_id`
+
+	transfer := &sqlTransfer{
+		CustomerID: customerID,
+		TaskID:     fhirTaskID,
+	}
+
+	_, err = tx.NamedExecContext(ctx, query, transfer)
```

Comment:
I think you should use `tx.GetContext` for getting a value. This also gives back the `sql.ErrNoRows`.